misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractIterableNotToContainValuesExpectationsTest.kt

```diff
@@ -0,0 +1,178 @@
+package ch.tutteli.atrium.specs.integration
+
+import ch.tutteli.atrium.api.fluent.en_GB.*
+import ch.tutteli.atrium.api.verbs.internal.expect
+import ch.tutteli.atrium.core.polyfills.format
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.specs.*
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.afterExplanatoryIndent
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.afterMismatchedWarning
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.hasANextElement
+import ch.tutteli.atrium.specs.integration.IterableToContainEntriesSpecBase.Companion.mismatchedIndex
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.anElementWhichEquals
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.fluentEmpty
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.mismatches
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.notToContainDescr
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.oneToSeven
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.oneToSevenNullable
+import ch.tutteli.atrium.specs.integration.IterableToContainSpecBase.Companion.separator
+import ch.tutteli.atrium.specs.integration.utils.SubjectLessTestData
+import ch.tutteli.atrium.testfactories.TestFactory
+import ch.tutteli.atrium.translations.DescriptionIterableLikeExpectation
+
+@Suppress("FunctionName")
+abstract class AbstractIterableNotToContainValuesExpectationsTest(
+    private val notToContainValuesSpec: Fun2<Iterable<Double>, Double, Array<out Double>>,
+    private val notToContainNullableValuesSpec: Fun2<Iterable<Double?>, Double?, Array<out Double?>>,
+    private val notToHaveElementsOrNoneFunNameSpec: String,
+) : ExpectationFunctionBaseTest() {
+
+    @TestFactory
+    fun subjectLessTest() = subjectLessTestFactory(
+        SubjectLessTestData(
+            notToContainValuesSpec.forSubjectLessTest(2.3, arrayOf())
+        ),
+        SubjectLessTestData(
+            notToContainNullableValuesSpec.forSubjectLessTest(2.3, arrayOf())
+        )
+    )
+
+
+    fun Expect<Iterable<Double?>>.notToContainNullableFun(a: Double?, vararg aX: Double?) =
+        notToContainNullableValuesSpec(this, a, aX)
+
+    val anElementWhichIsWithIndent = "$indentRootBulletPoint$listBulletPoint$anElementWhichEquals"
+
+
+    @TestFactory
+    fun empty_collection__4_throws() = nonNullableCases(
```

Comment:
Doesn't matter that much what we expect, it will always throw
```suggestion
    fun empty_collection__throws() = nonNullableCases(
```

Reply:
@robstoll - Done.

Reply:
@robstoll - Done.